// @mendix/pluggable-widgets-tools/dist/typings-generator/generatePreviewTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generatePreviewTypes = void 0;
const helpers_1 = require("./helpers");
function generatePreviewTypes(widgetName, properties) {
    const results = Array.of();
    results.push(`export interface ${widgetName}PreviewProps {
    class: string;
    style: string;
${generatePreviewTypeBody(properties, results)}
}`);
    return results;
}
exports.generatePreviewTypes = generatePreviewTypes;
function generatePreviewTypeBody(properties, generatedTypes) {
    return properties.map(prop => `    ${prop.$.key}: ${toPreviewPropType(prop, generatedTypes)};`).join("\n");
}
function toPreviewPropType(prop, generatedTypes) {
    var _a;
    switch (prop.$.type) {
        case "boolean":
            return "boolean";
        case "string":
            return "string";
        case "action":
            return "{} | null";
        case "textTemplate":
            return "string";
        case "integer":
        case "decimal":
            return "number | null";
        case "icon":
            return '{ type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null';
        case "image":
            return "string";
        case "file":
            return "string";
        case "datasource":
            return "{} | null";
        case "attribute":
        case "expression":
            return "string";
        case "enumeration":
            return helpers_1.capitalizeFirstLetter(prop.$.key) + "Enum";
        case "object":
            if (!((_a = prop.properties) === null || _a === void 0 ? void 0 : _a.length)) {
                throw new Error("[XML] Object property requires properties element");
            }
            const childType = helpers_1.capitalizeFirstLetter(prop.$.key) + "PreviewType";
            generatedTypes.push(`export interface ${childType} {
${generatePreviewTypeBody(helpers_1.extractProperties(prop.properties[0]), generatedTypes)}
}`);
            return prop.$.isList === "true" ? `${childType}[]` : childType;
        case "widgets":
            return "{ widgetCount: number; renderer: ComponentType }";
        default:
            return "any";
    }
}
